src/server.ts

```diff
@@ -1,37 +1,43 @@
-import 'reflect-metadata';
+import * as express from 'express';
 import { ApolloServer } from 'apollo-server-express';
-import express from 'express';
 import { buildSchema } from 'type-graphql';
-import { HelloResolver } from './hello.resolver';
+import { ConnectionOptions, createConnection } from 'typeorm';
+import { Service } from 'typedi';
+import { Container } from 'typedi';
 
+@Service()
 export class Server {
-	private readonly app = express();
-	private readonly DEFAULT_PORT = parseInt(process.env.PORT) || 4000;
-
-	constructor() {
-		this.run();
-	}
-
-	private async run() {
-		const schema = await buildSchema({
-			resolvers: [HelloResolver],
-			validate: false,
-		});
-
-		const server = new ApolloServer({
-			context: ({ req, res }) => ({
-				req,
-				res,
-			}),
-			schema,
-		});
-
-		server.applyMiddleware({ app: this.app });
-	}
-
-	public listen(callback: (port: number) => void): void {
-		this.app.listen(this.DEFAULT_PORT, () => {
-			callback(this.DEFAULT_PORT);
-		});
-	}
+  public async run() {
+    const schema = await buildSchema({
+      resolvers: [__dirname + '/**/*.resolver{.ts,.js}'],
+      validate: false,
+      container: Container,
+    });
+
+    const port = +(process.env.PORT || 4000);
+
+    const server = new ApolloServer({
+      schema,
+    });
+
+    const app = express();
+
+    server.applyMiddleware({ app });
+
+    return app.listen({ port }, () => {
+      console.log(`server running at port: ${port}`);
+    });
+  }
+
+  public async connect() {
```

Comment:
```suggestion
  public async connectDb() {
```

Pra ficar mais claro o que esse métod faz/conecta.